### PR TITLE
Enables pedantic warnings and 'treat warnings as errors'.

### PR DESCRIPTION
**Pull request by StevenEWright**
This change is [Reviewable](https://reviewable.io/reviews/openid/appauth-ios/62)

### CODE REVIEW

**Comment by StevenEWright** [comment]
@WilliamDenniss I can't see what the bitrise error is. Can you tell me?

**Comment by WilliamDenniss** [comment]
Reviewed 12 of 12 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[Source/iOS/OIDAuthorizationUICoordinatorIOS.h, line 37 at r1](https://reviewable.io:443/reviews/openid/appauth-ios/62#-KXKVwErdd9APjGfyBxT:-KXKVwErdd9APjGfyBxU:bbj8vfa) ([raw file](https://github.com/openid/appauth-ios/blob/ace857fb6679a2d52119abf0be340bc0b611d328/Source/iOS/OIDAuthorizationUICoordinatorIOS.h#L37)):*
> ```objc
> /*! @brief The designated initializer.
>     @param presentingViewController The view controller from which to present the
>         \SFSafariViewController.
> ```

This slash is a Doxygen macro that will like out to Apple's docs. Did you intend to remove it?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openid/appauth-ios/62)*

**Comment by WilliamDenniss** [comment]
Errors from bitrise:

```
▸ Building AppAuth/AppAuth [Debug]
▸ Check Dependencies
▸ Compiling OIDErrorUtilities.m
▸ Compiling OIDTokenUtilities.m
▸ Compiling OIDGrantTypes.m
▸ Compiling OIDTokenRequest.m
▸ Compiling OIDAuthState+IOS.m
▸ Compiling OIDTokenResponse.m
▸ Compiling OIDScopeUtilities.m
▸ Compiling OIDAuthorizationResponse.m
▸ Compiling OIDServiceConfiguration.m
▸ Compiling OIDAuthState.m
▸ Compiling OIDAuthorizationService.m
▸ Compiling OIDAuthorizationUICoordinatorIOS.m
▸ Compiling OIDURLQueryComponent.m
▸ Compiling OIDFieldMapping.m
▸ Compiling OIDError.m
▸ Compiling OIDAuthorizationRequest.m
▸ Compiling OIDAuthorizationService+IOS.m
▸ Compiling OIDResponseTypes.m
▸ Compiling OIDScopes.m
▸ Compiling OIDServiceDiscovery.m
▸ Building library libAppAuth.a
▸ Build Succeeded
▸ Building AppAuth/AppAuth [Debug]
▸ Check Dependencies
▸ Building AppAuth/AppAuthTests [Debug]
▸ Check Dependencies
▸ Processing UnitTestsInfo.plist
▸ Compiling OIDURLQueryComponentTests.m
▸ Compiling OIDTokenResponseTests.m

❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:136:35: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertNotNil(response.request);
                                  ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:137:68: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(response.accessToken, kAccessTokenTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:138:64: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(response.tokenType, kTokenTypeTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:139:60: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(response.idToken, kIDTokenTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:140:70: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(response.refreshToken, kRefreshTokenTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:141:57: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(response.scope, kScopesTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:143:54: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

                        kTestAdditionalParameterValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:149:86: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssert(expiration > kExpiresInTestValue - 5 && expiration <= kExpiresInTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:153:39: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertNotNil(responseCopy.request);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:154:72: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(responseCopy.accessToken, kAccessTokenTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:155:68: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(responseCopy.tokenType, kTokenTypeTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:156:64: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(responseCopy.idToken, kIDTokenTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:157:74: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(responseCopy.refreshToken, kRefreshTokenTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:158:61: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(responseCopy.scope, kScopesTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:160:54: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

                        kTestAdditionalParameterValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:175:39: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertNotNil(responseCopy.request);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:176:81: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(responseCopy.request.clientID, response.request.clientID);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:178:72: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(responseCopy.accessToken, kAccessTokenTestValue);
        ^



❌  /Users/vagrant/git/UnitTests/OIDTokenResponseTests.m:179:68: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  XCTAssertEqualObjects(responseCopy.tokenType, kTokenTypeTestValue);
        ^



❌  fatal error: too many errors emitted, stopping now [-ferror-limit=]


xcode test exit code: 65
xcode test failed, error: exit status 65
exit status 1
|

**Comment by StevenEWright** [comment]
Wonderful! Thank you! On it...

---

Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks failed.

---

*[Source/iOS/OIDAuthorizationUICoordinatorIOS.h, line 37 at r1](https://reviewable.io:443/reviews/openid/appauth-ios/62#-KXKVwErdd9APjGfyBxT:-KXMgx6RxXpC_yuwtiO9:b8yp00s) ([raw file](https://github.com/openid/appauth-ios/blob/ace857fb6679a2d52119abf0be340bc0b611d328/Source/iOS/OIDAuthorizationUICoordinatorIOS.h#L37)):*
<details><summary><i>Previously, WilliamDenniss (William Denniss) wrote…</i></summary>
> This slash is a Doxygen macro that will like out to Apple's docs. Did you intend to remove it?
</details>
No I did not, in that case!! Thank you!!! Will fix...

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openid/appauth-ios/62#-:-KXMgpbJOQIz1WC6qzj2:b-wsb4eq)*

**Comment by StevenEWright** [comment]
Review status: 10 of 24 files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/iOS/OIDAuthorizationUICoordinatorIOS.h, line 37 at r1](https://reviewable.io:443/reviews/openid/appauth-ios/62#-KXKVwErdd9APjGfyBxT:-KXldkD5unrG3bomF7z-:b-896fix) ([raw file](https://github.com/openid/appauth-ios/blob/ace857fb6679a2d52119abf0be340bc0b611d328/Source/iOS/OIDAuthorizationUICoordinatorIOS.h#L37)):*
<details><summary><i>Previously, StevenEWright (Steven E Wright) wrote…</i></summary><blockquote>

No I did not, in that case!! Thank you!!! Will fix...
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/openid/appauth-ios/62)*

**Comment by WilliamDenniss** [comment]
Bitrise shows 1 error:
```
 Compiling OIDAuthState.m

❌  /Users/vagrant/git/Source/OIDAuthState.m:421:81: must specify at least one argument for '...' parameter of variadic macro [-Werror,-Wgnu-zero-variadic-macro-arguments]

  NSAssert(_pendingActionsSyncObject, @"_pendingActionsSyncObject cannot be nil");
```

**Comment by WilliamDenniss** [comment]
#245 is a rebase of this PR with additional fixes.

File: Source/OIDAuthState.m [hunk 1]
```diff
@@ -283,7 +282,7 @@ - (BOOL)isAuthorized {
 - (void)updateWithAuthorizationResponse:(nullable OIDAuthorizationResponse *)authorizationResponse
                                   error:(nullable NSError *)error {
   // If the error is an OAuth authorization error, updates the state. Other errors are ignored.
-  if (error.domain == OIDOAuthAuthorizationErrorDomain) {
+  if (error && error.domain == OIDOAuthAuthorizationErrorDomain) {
```
Comment (WilliamDenniss) [inline Source/OIDAuthState.m]: nil check added here

**Pull request closed without merging**